### DISCUSSION TITLE
Lose bold style of symbols using symbol_map

### DISCUSSION THREAD

**Discussion by alohaia** [kovidgoyal/kitty#5119] [Q&A]
I use `symbol_map` to set the font for Chinese characters:

```
symbol_map U+4E00-U+9FFF,U+3400-U+4DBF,U+20000-U+2A6DF,U+2A700–U+2B73F,U+2B740–U+2B81F,U+2B820–U+2CEAF,U+F900-U+FAFF,U+2F800-U+2FA1F Noto Sans CJK SC
```

But the "bold" Chinese characters are not bold.

![Screenshot_20220523_223627](https://user-images.githubusercontent.com/36324537/169846853-9cd75e34-4ee4-4093-95be-05f4635f06cb.png)

It seems that kitty only loads the regular font for `symbol_map`:

```
❯ kitty --debug-font-fallback
[143 23:06:11.435145] Preloaded font faces:
[143 23:06:11.435176] normal face: /usr/share/fonts/nerd-fonts/JetBrains Mono Regular Nerd Font Complete.ttf:0
[143 23:06:11.435185] bold face: /usr/share/fonts/nerd-fonts/JetBrains Mono Bold Nerd Font Complete.ttf:0
[143 23:06:11.435192] italic face: /usr/share/fonts/nerd-fonts/JetBrains Mono Italic Nerd FontComplete.ttf:0
[143 23:06:11.435199] bi face: /usr/share/fonts/nerd-fonts/JetBrains Mono Bold Italic Nerd Font Complete.ttf:0
[143 23:06:11.435208] Symbol map faces:
[143 23:06:11.435218] /usr/share/fonts/TTF/Hack-Regular.ttf:0
[143 23:06:11.435225] /usr/share/fonts/noto-cjk/NotoSansCJK-Regular.ttc:2
❯ fc-list "Noto Sans CJK SC"
/usr/share/fonts/noto-cjk/NotoSansCJK-Regular.ttc: Noto Sans CJK SC:style=Regular
/usr/share/fonts/noto-cjk/NotoSansCJK-Light.ttc: Noto Sans CJK SC,Noto Sans CJK SC Light:style=Light,Regular
/usr/share/fonts/noto-cjk/NotoSansCJK-Thin.ttc: Noto Sans CJK SC,Noto Sans CJK SC Thin:style=Thin,Regular
/usr/share/fonts/noto-cjk/NotoSansCJK-Black.ttc: Noto Sans CJK SC,Noto Sans CJK SC Black:style=Black,Regular
/usr/share/fonts/noto-cjk/NotoSansCJK-DemiLight.ttc: Noto Sans CJK SC,Noto Sans CJK SC DemiLight:style=DemiLight,Regular
/usr/share/fonts/noto-cjk/NotoSansCJK-Medium.ttc: Noto Sans CJK SC,Noto Sans CJK SC Medium:style=Medium,Regular
/usr/share/fonts/noto-cjk/NotoSansCJK-Bold.ttc: Noto Sans CJK SC:style=Bold
```

**Comment by kovidgoyal** [comment]
yes, that's by design. symbol_map is meant to be used for symbols not
regular text. If you do use it for regular text you lose formatting
possibilities such as bold/italic


On Mon, May 23, 2022 at 08:09:28AM -0700, aloha wrote:
> I use `symbol_map` to set the font for Chinese characters:
> 
> ```
> symbol_map U+4E00-U+9FFF,U+3400-U+4DBF,U+20000-U+2A6DF,U+2A700–U+2B73F,U+2B740–U+2B81F,U+2B820–U+2CEAF,U+F900-U+FAFF,U+2F800-U+2FA1F Noto Sans CJK SC
> ```
> 
> But the "bold" Chinese characters are not bold.
> 
> ![Screenshot_20220523_223627](https://user-images.githubusercontent.com/36324537/169846853-9cd75e34-4ee4-4093-95be-05f4635f06cb.png)
> 
> It seems that kitty only loads the regular font for `symbol_map`:
> 
> ```
> ❯ kitty --debug-font-fallback
> [143 23:06:11.435145] Preloaded font faces:
> [143 23:06:11.435176] normal face: /usr/share/fonts/nerd-fonts/JetBrains Mono Regular Nerd Font Complete.ttf:0
> [143 23:06:11.435185] bold face: /usr/share/fonts/nerd-fonts/JetBrains Mono Bold Nerd Font Complete.ttf:0
> [143 23:06:11.435192] italic face: /usr/share/fonts/nerd-fonts/JetBrains Mono Italic Nerd FontComplete.ttf:0
> [143 23:06:11.435199] bi face: /usr/share/fonts/nerd-fonts/JetBrains Mono Bold Italic Nerd Font Complete.ttf:0
> [143 23:06:11.435208] Symbol map faces:
> [143 23:06:11.435218] /usr/share/fonts/TTF/Hack-Regular.ttf:0
> [143 23:06:11.435225] /usr/share/fonts/noto-cjk/NotoSansCJK-Regular.ttc:2
> ❯ fc-list "Noto Sans CJK SC"
> /usr/share/fonts/noto-cjk/NotoSansCJK-Regular.ttc: Noto Sans CJK SC:style=Regular
> /usr/share/fonts/noto-cjk/NotoSansCJK-Light.ttc: Noto Sans CJK SC,Noto Sans CJK SC Light:style=Light,Regular
> /usr/share/fonts/noto-cjk/NotoSansCJK-Thin.ttc: Noto Sans CJK SC,Noto Sans CJK SC Thin:style=Thin,Regular
> /usr/share/fonts/noto-cjk/NotoSansCJK-Black.ttc: Noto Sans CJK SC,Noto Sans CJK SC Black:style=Black,Regular
> /usr/share/fonts/noto-cjk/NotoSansCJK-DemiLight.ttc: Noto Sans CJK SC,Noto Sans CJK SC DemiLight:style=DemiLight,Regular
> /usr/share/fonts/noto-cjk/NotoSansCJK-Medium.ttc: Noto Sans CJK SC,Noto Sans CJK SC Medium:style=Medium,Regular
> /usr/share/fonts/noto-cjk/NotoSansCJK-Bold.ttc: Noto Sans CJK SC:style=Bold
> ```
> 
> 
> 
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/kovidgoyal/kitty/discussions/5119
> You are receiving this because you are subscribed to this thread.
> 
> Message ID: ***@***.***>

-- 
_____________________________________

Dr. Kovid Goyal 
https://www.kovidgoyal.net
https://calibre-ebook.com
_____________________________________

**Comment by jediaxe** [comment]
Is there a workaround for this? This looks like a limitation to me, If fixing it doesn't break anything I say it's worth fixing.

The lower case L (l) is too close to the number 1 in the font I am using. This actually cost us a couple of days of troubleshooting because we missed the typographical error.

It would be ideal if the symbol_map works for bold too :)
```
# l too close to number 1
symbol_map U+006C firacode Nerd Font
# l Bold too close to number 1 Bold
symbol_map U+1D41C firacode Nerd Font Bold
```

**Comment by kovidgoyal** [comment]
This kind of thing should be fixed in the font, you can either patch it yourself or request the font developer to patch it.
